codec-http/src/main/java/io/netty/handler/codec/http/DefaultHttpHeaders.java

```diff
@@ -53,6 +53,9 @@ public boolean process(byte value) throws Exception {
     static final NameValidator<CharSequence> HttpNameValidator = new NameValidator<CharSequence>() {
         @Override
         public void validateName(CharSequence name) {
+            if (name == null || name.length() == 0) {
+                throw new IllegalArgumentException("empty headers are not allowed [" + name + "]");
```

Comment:
`PlatformDependent.throwException` is a hack to allow checked exceptions to be thrown when the signature does not allow for it. The `validateName` signature does not allow for any checked exceptions to be thrown.

`IllegalArgumentException` is an unchecked exception so it can be thrown regardless of method signature.